x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/CcsSlowLogRestIT.java

```diff
@@ -0,0 +1,358 @@
+            putRunAsUserRequest.setJsonEntity("""
+                {
+                  "password": "x-pack-test-password",
+                  "roles": ["run_as_remote_role"],
+                  "full_name": "Run As User"
+                }""");
+            assertOK(adminClient().performRequest(putRunAsUserRequest));
+
+            // Create target user (who will be run as)
+            final var putTargetUserRequest = new Request("PUT", "/_security/user/target_user");
+            putTargetUserRequest.setJsonEntity("""
+                {
+                  "password": "x-pack-test-password",
+                  "roles": ["run_as_remote_role"],
+                  "full_name": "Target User"
+                }""");
+            assertOK(adminClient().performRequest(putTargetUserRequest));
+
+            // Perform search with run-as header
+            final var runAsSearchRequest = new Request("GET", "/my_remote_cluster:run_as_test/_search");
+            runAsSearchRequest.setJsonEntity("""
+                {
+                  "query": { "match_all": {} }
+                }""");
+
+            // Add both authentication and run-as headers
+            runAsSearchRequest.setOptions(
+                RequestOptions.DEFAULT.toBuilder()
+                    .addHeader("Authorization", basicAuthHeaderValue("run_as_user", PASS))
+                    .addHeader("es-security-runas-user", "target_user")
+            );
+
+            final Response runAsResponse = client().performRequest(runAsSearchRequest);
+            assertOK(runAsResponse);
+
+            // Verify slow log shows both authenticating and effective users from querying cluster
+            Map<String, Object> expectedRunAsAuthContext = Map.of(
+                "user.name",
+                "run_as_user",                    // Authenticating user from querying cluster
+                "user.realm",
+                "default_native",
+                "user.full_name",
+                "Run As User",
+                "user.effective.name",
+                "target_user",          // Effective user from querying cluster
+                "user.effective.realm",
+                "default_native",
+                "user.effective.full_name",
+                "Target User",
+                "auth.type",
+                "REALM"
+            );
+
+            verifySlowLogAuthenticationContext(expectedRunAsAuthContext);
+        }
+    }
+
+    /**
+     * Verifies that the slow logs on the fulfilling cluster contain the expected
+     * authentication context from the original user on the querying cluster.
+     */
+    private void verifySlowLogAuthenticationContext(Map<String, Object> expectedAuthContext) throws Exception {
+        assertBusy(() -> {
+            try (var slowLog = fulfillingCluster.getNodeLog(0, LogType.SEARCH_SLOW)) {
+                final List<String> lines = Streams.readAllLines(slowLog);
+                assert (!lines.isEmpty());
```

Comment:
Ah surprised checkStyle is not catching this -- ES style guides insist on using `false == lines.isEmpty()` instead of negation. Lets follow that here, too (FWIW, I also prefer the `!` approach but 'tis the check-style enforced convention)

Reply:
or actually `assertThat(lines, not(empty()));` probs best here